### PR TITLE
Add e2e example for `Machine` creation using ephemeral resources

### LINKED ISSUE #1129: E2e examples
# Summary

Create a library of sample e2e tests in the `config/samples/e2e` directory covering the following scenarios. Make sure those samples can be customized via `kustomization` e.g. for using custom `MachineClasses` and `VolumeClasses`.

## Scenarios

- [x] Create simple `Machine` with **non-ephemeral** `Volume` and `NetworkInterface` to install nginx and expose it to the internet via a `VirtualIP`
- [x] Create simple `Machine` with **ephemeral** `Volume` and `NetworkInterface` to install nginx and expose it to the internet via a `VirtualIP`
- [x] IPAM examples illustrating the `Prefix` allocation for both IPv4 and IPv6. The e2e samples should also how the sub `Prefix` allocation.
- [x] Create example illustrating `LoadBalancer` with 2 simple `Machines`, `NetworkInterface` pointing to same Network and `LoadBalancer` with label selector for created `NetworkInterface`
- [x] Create simple example illustrating 'Network' peering with two `Network`s peered to each other
- [ ] Illustrate `NetworkPolicy` example to allow ssh at port 22 to a machine from another machine

### PR DESCRIPTION
**Pull request by Rohit-0505**
# Proposed Changes

- Add e2e example manifests for `machine` creation using ephemeral `volume` and `networkinterface`

/ref #1129

### CODE REVIEW

**Review by afritzler — CHANGES_REQUESTED** [review]
Can we name the sample folder `machine-with-ephemeral` resources? This naming scheme would fit better to the existing e2e examples.

**Comment by Rohit-0505** [comment]
> Can we name the sample folder `machine-with-ephemeral` resources? This naming scheme would fit better to the existing e2e examples.

done

File: config/samples/e2e/bases/machine-with-ephemeral-resources/machine.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          networkRef:
+            name: network-sample
+          virtualIP:
+            ephemeral:
+              virtualIPTemplate:
+                spec:
+                  type: Public
+                  ipFamily: IPv4
+  volumes:
+  - name: root-disk-1
```
Comment (afritzler) [inline config/samples/e2e/bases/machine-with-ephemeral-resources/machine.yaml:29]: Can we name this volume `root-disk`.

File: config/samples/e2e/machine-with-ephemeral-resources/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+kind: Machine
+metadata:
+  name: machine-sample
+spec:
+  machineClassRef:
+    name: new-machineClass
+  machinePoolRef:
+    name: new-machinePool
+  volumes:
+  - name: root-disk-1
```
Comment (afritzler) [inline config/samples/e2e/machine-with-ephemeral-resources/README.md:31]: `root-disk`

**Review by afritzler — CHANGES_REQUESTED** [review]

**Review by afritzler — APPROVED** [review]

**Pull request merged**